cns/client.go

```diff
@@ -190,11 +190,11 @@ func (c *Client) QueryAllVolume(ctx context.Context, queryFilter cnstypes.CnsQue
 }
 
 // QueryVolumeAsync calls the CNS QueryAsync API and return a task, from which we can extract CnsQueryResult
-func (c *Client) QueryVolumeAsync(ctx context.Context, queryFilter cnstypes.CnsQueryFilter, querySelection cnstypes.CnsQuerySelection) (*object.Task, error) {
+func (c *Client) QueryVolumeAsync(ctx context.Context, queryFilter cnstypes.CnsQueryFilter, querySelection *cnstypes.CnsQuerySelection) (*object.Task, error) {
```

Comment:
So, using`nil` or `&CnsQuerySelection{}` as the input of `querySelection` will end up with different QueryResults, right? Is the logic handled at CNS layer?

Reply:
Yes. The logic is handled in CNS layer. That's correct